ceph/rados/core_workflows.py

```diff
@@ -2740,3 +2740,87 @@ def get_host_object(self, hostname):
 
         log.error(f"Could not find host object for host {hostname}")
         return None
+
+    def get_object_details_head(self, osd_id, object_name):
+        """
+        Method returns the object head in json format
+        Args:
+            osd_id: osd id number
+            object_name : object name
+        Return : Returns the object head details if exists or None
+
+        """
+        cmd_get_object = (
+            f"cephadm shell --name osd.{osd_id} -- ceph-objectstore-tool --data-path "
+            f"/var/lib/ceph/osd/ceph-{osd_id} --head --op list {object_name}"
+        )
+        acting_osd_node = self.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        out_put = acting_osd_node.exec_command(sudo=True, cmd=cmd_get_object)
+        object_head = out_put[0].strip()
+        if object_name not in object_head:
+            log.error(
+                f"Not able to retrive the object head for the object-{object_name} "
+            )
+            return None
+        log.info(f"The object head of the {object_name} is {object_head}")
+        return object_head
+
+    def set_snapset_corrupt(self, osd_id, object_head):
+        """
+        Method to clear the snapset
+        Args:
+            osd_id : osd id
+            object_head: object head in json format
+        Return: None
+        """
+        cmd_snapset_corrupt = (
+            f"cephadm shell --name osd.{osd_id} -- ceph-objectstore-tool --data-path"
+            f" /var/lib/ceph/osd/ceph-{osd_id}  '{object_head}'   clear-snapset corrupt"
+        )
+        cmd_snapset_corrupt = cmd_snapset_corrupt.replace("\\", "")
+        acting_osd_node = self.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        acting_osd_node.exec_command(sudo=True, cmd=cmd_snapset_corrupt)
+        return None
+
+    def create_inconsistent_obj_snap(self, pool_name, object_name):
```

Comment:
please add details about what the method would return in case of both pass/Fail scenarios.

Reply:
Code  modified.